typescript/phoenix-sdk/src/marketMetadata.ts

```diff
@@ -0,0 +1,818 @@
+import { MarketState } from "./market";
+import { MarketSizeParams, OrderPacket, TokenParams } from "./types";
+import { getSeatAddress, toNum } from "./utils";
+import { PublicKey, TransactionInstruction } from "@solana/web3.js";
+import { getAssociatedTokenAddressSync } from "@solana/spl-token";
+import {
+  CancelMultipleOrdersByIdInstructionArgs,
+  CancelMultipleOrdersByIdWithFreeFundsInstructionArgs,
+  CancelUpToInstructionArgs,
+  CancelUpToWithFreeFundsInstructionArgs,
+  DepositFundsInstructionArgs,
+  LimitOrderTemplate,
+  PROGRAM_ID,
+  PlaceMultiplePostOnlyOrdersInstructionArgs,
+  PostOnlyOrderTemplate,
+  ImmediateOrCancelOrderTemplate,
+  ReduceOrderInstructionArgs,
+  ReduceOrderWithFreeFundsInstructionArgs,
+  WithdrawFundsInstructionArgs,
+  createCancelAllOrdersInstruction,
+  createCancelAllOrdersWithFreeFundsInstruction,
+  createCancelMultipleOrdersByIdInstruction,
+  createCancelMultipleOrdersByIdWithFreeFundsInstruction,
+  createCancelUpToInstruction,
+  createCancelUpToWithFreeFundsInstruction,
+  createDepositFundsInstruction,
+  createPlaceLimitOrderInstruction,
+  createPlaceLimitOrderWithFreeFundsInstruction,
+  createPlaceMultiplePostOnlyOrdersInstruction,
+  createPlaceMultiplePostOnlyOrdersWithFreeFundsInstruction,
+  createReduceOrderInstruction,
+  createReduceOrderWithFreeFundsInstruction,
+  createRequestSeatInstruction,
+  createSwapInstruction,
+  createSwapWithFreeFundsInstruction,
+  createWithdrawFundsInstruction,
+  getImmediateOrCancelOrderPacket,
+  getLimitOrderPacket,
+  getLogAuthority,
+  getPostOnlyOrderPacket,
+} from "./index";
+
+export class MarketMetadata {
+  address: PublicKey;
+  marketSizeParams: MarketSizeParams;
+  baseParams: TokenParams;
+  baseLotSize: number;
+  quoteParams: TokenParams;
+  quoteLotSize: number;
+  tickSizeInQuoteAtomsPerBaseUnit: number;
+  rawBaseUnitsPerBaseUnit: number;
+  // The number of base lots per base unit
+  baseLotsPerBaseUnit: number;
+  // Tick size of the market, in quote lots per base unit
+  // Note that the header contains tick size in quote atoms per base unit
+  quoteLotsPerBaseUnitPerTick: number;
```

Comment:
here would be a great place to copy-paste some of the docstrings

(also, newline between them once there's docs)